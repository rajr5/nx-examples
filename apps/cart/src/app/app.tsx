import React from 'react';

import { Route } from 'react-router-dom';

export const App = () => {
  return (
    <Route
      path="/cart"
      exact
      render={() => <div>Welcome to cart!</div>}
    />
  );
};

export default App;
