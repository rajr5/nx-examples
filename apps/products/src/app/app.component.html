Welcome to products!

<router-outlet></router-outlet>
