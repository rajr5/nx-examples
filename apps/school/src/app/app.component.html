<nx-header></nx-header>
<router-outlet></router-outlet>
