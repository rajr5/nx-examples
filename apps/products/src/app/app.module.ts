import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppComponent } from './app.component';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    RouterModule.forRoot(
      [
        {
          path: '',
          pathMatch: 'full',
          loadChildren: () =>
            import('@nx-example/products/home-page').then(
              module => module.ProductsHomePageModule
            )
        }
      ],
      { initialNavigation: 'enabled' }
    )
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule {}
