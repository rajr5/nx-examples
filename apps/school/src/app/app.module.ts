import { NgModule } from '@angular/core';
import { AppComponent } from './app.component';
import { BrowserModule } from '@angular/platform-browser';
import { NxModule } from '@nrwl/nx';
import { RouterModule } from '@angular/router';
import { schoolUiRoutes } from '@nx-examples/school-ui';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { environment } from '../environments/environment';
import { StoreRouterConnectingModule } from '@ngrx/router-store';
import { appInitialState, appReducer } from '@nx-examples/model';

@NgModule({
  imports: [
    BrowserModule,
    NxModule.forRoot(),
    RouterModule.forRoot(
      [
        { path: 'school-ui', children: schoolUiRoutes },
        { path: 'slides', loadChildren: '@nx-examples/slides#SlidesModule' }
      ],
      { initialNavigation: 'enabled' }
    ),
    StoreModule.forRoot(appReducer, {initialState: appInitialState}),
    EffectsModule.forRoot([]),
    !environment.production ? StoreDevtoolsModule.instrument() : [],
    StoreRouterConnectingModule
  ],
  declarations: [AppComponent],
  bootstrap: [AppComponent]
})
export class AppModule {}
