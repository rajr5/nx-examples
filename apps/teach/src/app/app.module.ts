import { NgModule } from '@angular/core';
import { AppComponent } from './app.component';
import { BrowserModule } from '@angular/platform-browser';
import { NxModule } from '@nrwl/nx';
import { RouterModule } from '@angular/router';
import { IntroComponent } from './intro/intro.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

@NgModule({
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    NxModule.forRoot(),
    RouterModule.forRoot(
      [
        { path: '', component: IntroComponent },
        { path: 'app1', loadChildren: '@nx-examples/teach/app1#TeachApp1Module' },
        { path: 'app2', loadChildren: '@nx-examples/teach/app2#TeachApp2Module' }
      ],
      { initialNavigation: 'enabled' }
    )
  ],
  declarations: [AppComponent, IntroComponent],
  bootstrap: [AppComponent]
})
export class AppModule {}
