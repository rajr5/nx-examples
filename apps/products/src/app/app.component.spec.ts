import { TestBed, async } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { RouterTestingModule } from '@angular/router/testing';
import { By } from '@angular/platform-browser';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [AppComponent]
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should render a title`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    expect(fixture.nativeElement.textContent).toContain('Welcome to products!');
  });

  it(`should render the route outlet`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    expect(fixture.nativeElement.querySelector('router-outlet')).toBeTruthy();
  });
});
