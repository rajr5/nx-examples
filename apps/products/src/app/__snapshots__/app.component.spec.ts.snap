// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<nx-example-root>
  Welcome to products! <router-outlet />
</nx-example-root>
`;
